src/bindings.c

```diff
@@ -1131,6 +1134,9 @@ int pr_namebind_open(const char *name, const pr_netaddr_t *addr) {
   }
 
   port = ntohs(pr_netaddr_get_port(addr));
+  if (port == 0) {
+    port = server_port;
+  }
```

Comment:
This is the key change; this `port == 0` check occurs in `pr_namebind_create()` as well, and probably _should_ have always been done here.